### ISSUE TITLE
test_frcnn.py not working?

### ISSUE THREAD

**Issue by tienthegainz** [you359/Keras-FasterRCNN#23]
The result_img didn't contain the bounding box. And when I print out true_x, true_y , .v..vv.... It just an empty list.

**Comment by 3ST4R** [comment]
I've also been facing the same problem, maybe the dataset is too small and the code is incapable of transfer learning

**Comment by gdut3118007086** [comment]
i use the voc2012, i've also been facing the same problem.

**Comment by 3ST4R** [comment]
I went into checking the code of this file; found out that the **bboxes** dict is empty, thereafter, running a **for** loop on it will be nullified (which contains the bbox plotting). I guess reducing bbox threshold may fix it, tho I haven't tried it yet.

**Comment by gdut3118007086** [comment]
I train VOC2012 epochs=126 loss rpn classifier = 3.22 oscillate around 3 without falling.

**Comment by gdut3118007086** [comment]
Maybe the data set is too big? Or is there a problem with the parameters of the training?

**Comment by gdut3118007086** [comment]
My model is currently only able to detect people and train for 8 hours. The current box is not very accurate

**Comment by 3ST4R** [comment]
Dataset is of 3471 images (originally of 267 images, increased by augmentation). Parameters of training are as default except I changed the learning rate to 0.01, and others optimizers, but same result.

**Comment by collinarnett** [comment]
@3ST4R I have tested this and it seems even if the bbox threshold is set to zero there still is no bbox.

**Comment by collinarnett** [comment]
I figured the issue out. The weight's are never loaded when doing inference

```
print('Loading weights from {}'.format(C.model_path))
model_rpn.load_weights(C.model_path, by_name=True)
model_classifier.load_weights(C.model_path, by_name=True)

model_rpn.compile(optimizer='sgd', loss='mse')
model_classifier.compile(optimizer='sgd', loss='mse')

all_imgs = []

classes = {}

bbox_threshold = 0.8

visualise = True

for idx, img_name in enumerate(sorted(os.listdir(img_path))):
    if not img_name.lower().endswith(('.bmp', '.jpeg', '.jpg', '.png', '.tif', '.tiff')):
        continue
    print(img_name)
    st = time.time()
    filepath = os.path.join(img_path,img_name)

    img = cv2.imread(filepath)

    X, ratio = format_img(img, C)

    if K.image_dim_ordering() == 'tf':
        X = np.transpose(X, (0, 2, 3, 1))

    # get the feature maps and output from the RPN
    [Y1, Y2, F] = model_rpn.predict(X)


    R = roi_helpers.rpn_to_roi(Y1, Y2, C, K.image_dim_ordering(), overlap_thresh=0.7)

    # convert from (x1,y1,x2,y2) to (x,y,w,h)
    R[:, 2] -= R[:, 0]
    R[:, 3] -= R[:, 1]

    # apply the spatial pyramid pooling to the proposed regions
    bboxes = {}
    probs = {}

    for jk in range(R.shape[0]//C.num_rois + 1):
        ROIs = np.expand_dims(R[C.num_rois*jk:C.num_rois*(jk+1), :], axis=0)
        if ROIs.shape[1] == 0:
            break

        if jk == R.shape[0]//C.num_rois:
            #pad R
            curr_shape = ROIs.shape
            target_shape = (curr_shape[0],C.num_rois,curr_shape[2])
            ROIs_padded = np.zeros(target_shape).astype(ROIs.dtype)
            ROIs_padded[:, :curr_shape[1], :] = ROIs
            ROIs_padded[0, curr_shape[1]:, :] = ROIs[0, 0, :]
            ROIs = ROIs_padded

        [P_cls, P_regr] = model_classifier_only.predict([F, ROIs])
```

`model_classifier_only` is used for inference but the weights are loaded for `model_classifier`

**Comment by 3ST4R** [comment]
@collinarnett I also checked your post about weights not loading, and I tried that, still there is no bbox.  

So, what should be a possible solution to this problem. As, I see the blogs where people getting it working absolutely fine, without any digging into the code, I wonder how.

**Comment by ljm52519** [comment]
@3ST4R have u fixed this problem? I'm facing this problem too.

**Comment by mkvashennikov** [comment]
> I train VOC2012 epochs=126 loss rpn classifier = 3.22 oscillate around 3 without falling.

@3ST4R, @collinarnett  im facing with this problem too. Maybe you fix this problem..?

**Comment by collinarnett** [comment]
@mkvashennikov I haven't looked a this since 2019. I don't remember fixing it. I think I used another implementation.
